spec/services/participants/defer/early_career_teacher_spec.rb

```diff
@@ -2,36 +2,103 @@
 
 require "rails_helper"
 
-require_relative "../../../shared/context/lead_provider_profiles_and_courses"
-
 RSpec.describe Participants::Defer::EarlyCareerTeacher do
-  include_context "lead provider profiles and courses"
-  let(:participant_params) do
-    {
-      cpd_lead_provider: cpd_lead_provider,
-      participant_id: ect_profile.user.id,
-      course_identifier: "ecf-induction",
-      reason: "career-break",
-    }
+  let(:cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider) }
+  let(:lead_provider) { cpd_lead_provider.lead_provider }
+  let(:profile) { create(:ect_participant_profile) }
+  let(:user) { profile.user }
+  let(:school) { profile.school_cohort.school }
+  let(:cohort) { profile.school_cohort.cohort }
+  let!(:partnership) do
+    create(
+      :partnership,
+      school: school,
+      lead_provider: lead_provider,
+      cohort: cohort,
+    )
+  end
+
+  subject do
+    described_class.new(
+      params: {
+        participant_id: user.id,
+        course_identifier: "ecf-induction",
+        cpd_lead_provider: cpd_lead_provider,
+        reason: "bereavement",
+      },
+    )
   end
 
-  it_behaves_like "a participant defer action service" do
-    def given_params
-      participant_params
+  describe "#call" do
+    it "updates profile training_status to deferred" do
+      expect { subject.call }.to change { profile.reload.training_status }.from("active").to("deferred")
     end
 
-    def user_profile
-      ect_profile.reload
+    it "creates a ParticipantProfileState" do
+      expect { subject.call }.to change { ParticipantProfileState.count }.by(1)
     end
-  end
 
-  it_behaves_like "a participant service for ect" do
-    def given_params
-      participant_params
+    context "when already deferred" do
+      before do
+        described_class.new(
+          params: {
+            participant_id: user.id,
+            course_identifier: "ecf-induction",
+            cpd_lead_provider: cpd_lead_provider,
+            reason: "bereavement",
+          },
+        ).call # must be different instance from subject
+      end
+
+      it "returns an error and does not update training_status" do
+        expect { subject.call }.to raise_error(ActiveRecord::RecordInvalid).and not_change { profile.reload.training_status }
+      end
+    end
+
+    context "when status is withdrawn" do
+      before do
+        ParticipantProfileState.create!(participant_profile: profile, state: "withdrawn")
+        profile.update!(status: "withdrawn")
+      end
+
+      xit "returns an error and does not update training_status" do
+        # TODO: there is a gap and bug here
+        # it should return a useful error
+        # but throws an error as we scope to active profiles only and therefore never find the record
```

Comment:
Could you add a ticket for this? sounds like something we want to address (with a link to this PR for context)